internal/controller/template_controller.go

```diff
@@ -359,19 +386,22 @@ func (r *TemplateReconciler) getHelmChartFromChartRef(ctx context.Context, chart
 	return helmChart, nil
 }
 
-func (r *ClusterTemplateReconciler) validateCompatibilityAttrs(ctx context.Context, template *kcm.ClusterTemplate) error {
-	management := new(kcm.Management)
-	if err := r.Client.Get(ctx, client.ObjectKey{Name: kcm.ManagementName}, management); err != nil {
+func (r *TemplateReconciler) getManagement(ctx context.Context, template templateCommon) (*kcm.Management, error) {
+	management := &kcm.Management{}
+	if err := r.Get(ctx, client.ObjectKey{Name: kcm.ManagementName}, management); err != nil {
 		if apierrors.IsNotFound(err) {
 			_ = r.updateStatus(ctx, template, "Waiting for Management creation to complete validation")
```

Comment:
I think now it would be better to move the template status update outside of the `getManagement` function, but it’s up to you.

Reply:
Discussed on slack. It’s not needed since the function is internal to the template reconciler, and updating the status within it is acceptable.